scripts/redirectURLParser.js

```diff
@@ -29,10 +31,16 @@ function langURLParser(state) {
 
   // invalid lang but with a valid path.
   if (!langTest && Object.keys(pages).indexOf(pathname) !== -1) {
-    return "/" + validLang + pathname;
+    pathname = "/" + validLang + pathname;
+  } else {
+    pathname = "/" + validLang + "/";
   }
-  // default
-  return "/" + validLang + "/";
```

Comment:
We are returning once at the end of the function now to reproduce less code related to query string values now being moved into this function and returned to make testing easier.